### DISCUSSION TITLE
Stop other Stacks

### DISCUSSION THREAD

**Discussion by southlakefll** [pybricks/support#1694] [Q&A]
If running multiple things simultaneously (say using nested multitask blocks), is there a way one task could end all of the other tasks (but keep the overall program still running)?

LEGO has a block called "stop other stacks", is there something similar in PyBricks?
![image](https://github.com/pybricks/support/assets/148405726/a402c284-0e89-4325-a960-35bf7d9cecaa)

**Comment by laurensvalk** [comment]
I think there would be a way to build it. Could you describe your use case so we can come up with an example?

Pybricks has a few useful ways of gracefully splitting and stopping parallel tasks.

This means there is usually a more elegant solution as compared to stopping all other tasks in an uncontrolled manner.

**Comment by southlakefll** [comment]
The specific use case I am thinking of is when using a multi-program switcher, with a menu (similar to your demo here: https://pybricks.com/project/spike-hub-menu/). But for when something goes wrong and the one program needs to be stopped. It would be nice to just stop that one thread and go back to the menu program.

For example, if say kids are running an FLL mission but it doesn't run correctly and they need to stop that mission in the middle by pressing the middle button (on Spike Prime). It would be nice for them to go back to the menu, at the same place, so they can potentially relaunch that same mission (without having to navigate the menu back to where they were).

I'm thinking now I might be able to work something up using the multitask, set to the "one task" mode. I haven't had a chance to try putting something together though. 

Thanks!

**Reply by laurensvalk** [reply]
> It would be nice to just stop that one thread and go back to the menu program.

There you have it! Stopping one thread does seem more elegant than the hammer approach of stopping everything :-)

You were on the right track with the multitask "until one task completes". You can use that to let either of these complete:
- a selected mission
- a wait for button press

... whichever completes first. This has the effect of running the mission until completion, unless the button is pressed first. I believe this is what you want.

![pybricks_program_2024-07-01T06_59_43](https://github.com/pybricks/support/assets/12326241/46127268-2555-4963-b637-e2447412e7d6)

Here is some basic code to get you started. I've left out the menu implementation. You can do that any way you like, and set the `selection` variable as the end result. It doesn't even have to be a full blown menu --- you could even set that variable by checking the positioning of an attachment, for example.
